src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4406,6 +4406,10 @@ private ParameterSyntax ParseParameter()
                         else
                         {
                             exclamationExclamation = MergeTokens(exclamationExclamation, this.EatToken(SyntaxKind.ExclamationToken), SyntaxKind.ExclamationExclamationToken);
```

Comment:
are you still intending to fix the missing diagnostic lifting for this case?

Reply:
ok.  so my recommendation is to get current token for being exclamation.  If so, grab and merge (if valid to do so).  

otherwise, have a dedicate branch for just single `!` and give the one messaget htere and convert it to !!.  then lower down you don't need to do any further checking that your !! was actually a !!.,